web/src/lib/components/shared-components/combobox.svelte

```diff
@@ -11,81 +11,204 @@
     selectedOption = option;
+    searchQuery = option.label;
     dispatch('select', option);
-    isOpen = false;
+    closeDropdown();
   };
 
   const onClear = () => {
     selectedOption = undefined;
     searchQuery = '';
     dispatch('select', selectedOption);
   };
+
+  const onClick = (event: MouseEvent) => {
+    const target = event.target as HTMLElement;
+    if (comboboxRef && !comboboxRef.contains(target)) {
+      deactivate();
+    }
+  };
 </script>
 
-<div class="relative w-full dark:text-gray-300 text-gray-700 text-base" use:clickOutside on:outclick={handleOutClick}>
+<label class="text-sm text-black dark:text-white" class:sr-only={hideLabel} for={inputId}>{label}</label>
+<div class="relative w-full dark:text-gray-300 text-gray-700 text-base" bind:this={comboboxRef}>
   <div>
-    {#if isOpen}
+    {#if isActive}
       <div class="absolute inset-y-0 left-0 flex items-center pl-3">
         <div class="dark:text-immich-dark-fg/75">
-          <Icon path={mdiMagnify} />
+          <Icon path={mdiMagnify} ariaHidden={true} />
         </div>
       </div>
     {/if}
 
     <input
-      {id}
       {placeholder}
-      role="combobox"
+      aria-activedescendant={selectedIndex || selectedIndex === 0 ? `${listboxId}-${selectedIndex}` : ''}
+      aria-autocomplete="list"
+      aria-controls={listboxId}
       aria-expanded={isOpen}
-      aria-controls={id}
-      class="immich-form-input text-sm text-left w-full !pr-12 transition-all"
-      class:!pl-8={isOpen}
+      autocomplete="off"
+      class:!pl-8={isActive}
       class:!rounded-b-none={isOpen}
-      class:cursor-pointer={!isOpen}
-      value={isOpen ? '' : selectedOption?.label || ''}
-      on:input={(e) => (searchQuery = e.currentTarget.value)}
-      on:focus={handleClick}
-      on:blur={() => (inputFocused = false)}
+      class:cursor-pointer={!isActive}
+      class="immich-form-input text-sm text-left w-full !pr-12 transition-all"
+      id={inputId}
+      on:click={activate}
+      on:focus={activate}
+      on:input={onInput}
+      role="combobox"
+      type="text"
+      value={searchQuery}
+      use:shortcuts={[
+        {
+          shortcut: { key: 'Tab' },
+          onShortcut: (event) => {
+            deactivate();
+            moveFocus(event);
+          },
+        },
+        {
+          shortcut: { key: 'Tab', shift: true },
+          onShortcut: (event) => {
+            deactivate();
+            moveFocus(event);
+          },
+        },
+        {
+          shortcut: { key: 'ArrowUp' },
+          onShortcut: () => {
+            openDropdown();
+            void incrementSelectedIndex(-1);
+          },
+        },
+        {
+          shortcut: { key: 'ArrowDown' },
+          onShortcut: () => {
+            openDropdown();
+            void incrementSelectedIndex(1);
+          },
+        },
+        {
+          shortcut: { key: 'ArrowDown', alt: true },
+          onShortcut: () => {
+            openDropdown();
```

Comment:
Is this shortcut even needed? The dropdown should always be opened when input has focus

Reply:
The suggestion to add this shortcut came from the WCAG spec:

https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-autocomplete-list/#kbd_label_textbox

There are scenarios where the input field is focused but the dropdown isn't visible, such as when the user presses Escape, or presses Enter without a dropdown completion highlighted. This just gives a quick way to open the dropdown again.